// @thirdweb-dev/wallets/evm/connectors/trust/dist/thirdweb-dev-wallets-evm-connectors-trust.browser.esm.js
import { InjectedConnector } from '../../injected/dist/thirdweb-dev-wallets-evm-connectors-injected.browser.esm.js';
import { a as assertWindowEthereum } from '../../../../dist/assertWindowEthereum-b48c1686.browser.esm.js';
import '../../../../dist/classPrivateFieldSet-a5db7c83.browser.esm.js';
import '../../../../dist/defineProperty-c8ecdc07.browser.esm.js';
import 'ethers';
import '../../../../dist/WagmiConnector-64a24de2.browser.esm.js';
import '@thirdweb-dev/chains';
import 'eventemitter3';
import '../../../../dist/normalizeChainId-e4cc0175.browser.esm.js';
import '../../../../dist/errors-4bdf81d2.browser.esm.js';

class TrustConnector extends InjectedConnector {
  constructor(arg) {
    const defaultOptions = {
      name: "Trust",
      getProvider() {
        function getReady(ethereum) {
          const isTrust = !!ethereum?.isTrust;
          if (!isTrust) {
            return;
          }
          return ethereum;
        }
        if (assertWindowEthereum(globalThis.window)) {
          if (globalThis.window.ethereum?.providers) {
            return globalThis.window.ethereum.providers.find(getReady);
          }
          return getReady(globalThis.window.ethereum);
        }
      }
    };
    const options = {
      ...defaultOptions,
      ...arg.options
    };
    super({
      chains: arg.chains,
      options,
      connectorStorage: arg.connectorStorage
    });
  }
}

export { TrustConnector };
